examples/vision/deeplabv3_plus.py

```diff
@@ -0,0 +1,298 @@
+
+    layer = tf.keras.layers.Concatenate(axis=-1)(
+        [out_pool, out_1, out_6, out_12, out_18]
+    )
+
+    output = convolution_block(layer, kernel_size=1)
+
+    return output
+
+
+"""
+The encoder features are first bilinearly upsampled by a factor of 4 and then 
+concatenated with the corresponding low-level features from the network backbone that
+have the same spatial resolution. For this example, we would be 
+using Resnet101 pre-trained on ImageNet as the backbone model and we would use 
+the low-level features from the Conv2 block of the backbone.
+"""
+
+
+def DeeplabV3Plus():
+    model_input = tf.keras.Input(shape=(IMAGE_SIZE, IMAGE_SIZE, 3))
+    resnet101 = tf.keras.applications.ResNet101(
+        weights='imagenet', include_top=False, input_tensor=model_input)
+    layer = resnet101.get_layer('conv4_block6_2_relu').output
+    layer = AtrousSpatialPyramidPooling(layer)
+    
+    input_a = tf.keras.layers.UpSampling2D(
+        size=(
+            IMAGE_SIZE // 4 // layer.shape[1],
+            IMAGE_SIZE // 4 // layer.shape[2]
+        ), interpolation = 'bilinear'
+    )(layer)
+    
+    input_b = resnet101.get_layer('conv2_block3_2_relu').output
+    input_b = convolution_block(input_b, n_filters=48, kernel_size=1)
+    
+    layer = tf.keras.layers.Concatenate(axis=-1)([input_a, input_b])
+    layer = convolution_block(layer)
+    layer = convolution_block(layer)
+    layer = tf.keras.layers.UpSampling2D(
+        size=(
+            IMAGE_SIZE // layer.shape[1],
+            IMAGE_SIZE // layer.shape[2]
+        ), interpolation='bilinear')(layer)
+    
+    model_output = tf.keras.layers.Conv2D(
+        NUM_CLASSES, kernel_size=(1, 1), padding='same')(layer)
+    
+    return tf.keras.Model(inputs=model_input, outputs=model_output)
+
+"""
+## Training
+"""
+
+"""
+We would train the model using Sparse Categorical Cross-entropy as the loss function and
+use Adam as the optimizer.
+"""
+
+loss = tf.losses.SparseCategoricalCrossentropy(from_logits=True)
+model.compile(
+    optimizer=tf.keras.optimizers.Adam(learning_rate=0.001),
+    loss=loss,
+    metrics=["accuracy"],
+)
+model.fit(dataset, epochs=25)
+
+plt.plot(history.history["loss"])
+plt.title("Training Loss")
+plt.ylabel("loss")
+plt.xlabel("epoch")
+plt.show()
+
+plt.plot(history.history["accuracy"])
+plt.title("Training Accuracy")
+plt.ylabel("accuracy")
+plt.xlabel("epoch")
+plt.show()
+
+"""
+## Inference using Colormap Overlay
+"""
+
+colormap = loadmat(
+    "./instance-level_human_parsing/instance-level_human_parsing/human_colormap.mat"
+)["colormap"]
+colormap = colormap * 100
+colormap = colormap.astype(np.uint8)
+
+
+def infer(model, image_tensor):
+    predictions = model.predict(np.expand_dims((image_tensor), axis=0))
+    predictions = np.squeeze(predictions)
+    predictions = np.argmax(predictions, axis=2)
+    return predictions
+
+
+def decode_segmask(mask, colormap, n_classes):
```

Comment:
decode_segmentation_masks -- in general use fully spelled out names everywhere

Reply:
Renamed `decode_segmask` to `decode_segmentation_masks`